### PR TITLE
stack renamed to valstack

### PR DESCRIPTION
**Pull request by QianyangPeng**
\<stack\> -> \<valstack\>
\</stack\> -> \</valstack\>
.Stack -> .ValStack
STACK -> VALSTACK

### CODE REVIEW

File: data.md [hunk 1]
```diff
@@ -194,34 +194,34 @@ Data Structures
...
+Operator `_++_` implements an append operator for sort `ValStack`.
 
 ```k
-    syntax Stack ::= ".Stack"
-                   | Val ":" Stack
-                   | Stack "++" Stack [function]
+    syntax ValStack ::= ".ValStack"
+                   | Val      ":"  ValStack
+                   | ValStack "++" ValStack [function]
  // --------------------------------------------
```
Comment (ehildenb) [inline data.md]: formatting

File: data.md [hunk 2]
```diff
@@ -194,34 +194,34 @@ Data Structures
...
 `#drop` will drop the prefix of a given stack, checking that the value types match the supplied type-sequence.
 
 ```k
-    syntax Stack ::= #zero ( ValTypes )         [function]
-                   | #take ( ValTypes , Stack ) [function]
-                   | #drop ( ValTypes , Stack ) [function]
+    syntax ValStack ::= #zero ( ValTypes )         [function]
+                   | #take ( ValTypes , ValStack ) [function]
+                   | #drop ( ValTypes , ValStack ) [function]
  // ------------------------------------------------------
```
Comment (ehildenb) [inline data.md]: formatting

File: media/memory-demo/wasm.k [hunk 3]
```diff
@@ -5,7 +5,7 @@ module WASM
```
Comment (ehildenb) [inline media/memory-demo/wasm.k]: This file probably shouldn't be changed, it should be saved in the state that @hjorthjort gave the presentation in I think.

File: test.md [hunk 4]
```diff
@@ -39,23 +39,23 @@ This asserts that a `trap` was just thrown.
...
 
-These functions make assertions about the state of the `<stack>` cell.
+These functions make assertions about the state of the `<valstack>` cell.
 
 ```k
     syntax Assertion ::= "#assertTopStack"        Val   String
                        | "#assertTopStackExactly" Val   String
-                       | "#assertStack"           Stack String
+                       | "#assertStack"           ValStack String
  // ----------------------------------------------------------
```
Comment (ehildenb) [inline test.md]: formatting

File: wasm.md [hunk 5]
```diff
@@ -372,10 +372,10 @@ Extension turns an `i32` type value into the corresponding `i64` type value.
     rule #convSourceType(extend_i32_s) => i32
 ```
 
-Stack Operations
+ValStack Operations
 ----------------
```
Comment (ehildenb) [inline wasm.md]: Formatting

File: wasm.md [hunk 6]
```diff
@@ -425,10 +425,10 @@ A block is the simplest way to create targets for break instructions (ie. jump d
 It simply executes the block then records a label with an empty continuation.
 
 ```k
-    syntax Label ::= "label" VecType "{" Instrs "}" Stack
+    syntax Label ::= "label" VecType "{" Instrs "}" ValStack
  // -----------------------------------------------------
```
Comment (ehildenb) [inline wasm.md]: formatting

File: wasm.md [hunk 7]
```diff
@@ -494,18 +494,18 @@ The various `init_local` variants assist in setting up the `locals` cell.
 
 ```k
     syntax Instr ::=  "init_local"  Int Val
-                   |  "init_locals"     Stack
-                   | "#init_locals" Int Stack
+                   |  "init_locals"     ValStack
+                   | "#init_locals" Int ValStack
  // -----------------------------------------
```
Comment (ehildenb) [inline wasm.md]: formatting

File: wasm.md [hunk 8]
```diff
@@ -635,21 +635,21 @@ Similar to labels, they sit on the instruction stack (the `<k>` cell), and `retu
 Unlike labels, only one frame can be "broken" through at a time.
 
 ```k
-    syntax Frame ::= "frame" ValTypes Stack Map
+    syntax Frame ::= "frame" ValTypes ValStack Map
  // -------------------------------------------
```
Comment (ehildenb) [inline wasm.md]: formatting

File: data.md [hunk 9]
```diff
@@ -194,34 +194,34 @@ Data Structures
...
+                   | #take ( ValTypes , ValStack ) [function]
+                   | #drop ( ValTypes , ValStack ) [function]
+ // ---------------------------------------------------------
+    rule #zero(.ValTypes)             => .ValStack
     rule #zero(ITYPE:IValType VTYPES) => < ITYPE > 0 : #zero(VTYPES)
 
-    rule #take(.ValTypes,   _)                           => .Stack
-    rule #take(TYPE VTYPES, < TYPE > VAL:Number : STACK) => < TYPE > VAL : #take(VTYPES, STACK)
+    rule #take(.ValTypes,   _)                           => .ValStack
+    rule #take(TYPE VTYPES, < TYPE > VAL:Number : VALSTACK) => < TYPE > VAL : #take(VTYPES, VALSTACK)
```
Comment (ehildenb) [inline data.md]: formatting (line up the `=>` signs).

Comment (QianyangPeng) [reply data.md]: fixed

**Pull request merged**